src/libraries/System.Private.CoreLib/src/Resources/Strings.resx

```diff
@@ -370,6 +370,9 @@
   <data name="Arg_HandleNotSync" xml:space="preserve">
     <value>Handle does not support synchronous operations. The parameters to the FileStream constructor may need to be changed to indicate that the handle was opened asynchronously (that is, it was opened explicitly for overlapped I/O).</value>
   </data>
+  <data name="Arg_HSCapacityOverflow" xml:space="preserve">
```

Comment:
I wanted to see what Dictionary does, and it doesn't have the same check (for attempting to resize beyond MaxPrimeArrayLength). At initialization time, that check doesn't seem too valuable (code that requests such a huge dictionary doesn't necessarily use it all) but I expect that if you try to add to many items to a dictionary it likely simply corrupts itself. (I wasn't able to try this as making a large enough Entries array would need many more GB than my laptop has). It might be worth an issue for Dictionary to get the same protection against corruption as HashSet.

Reply:
Actually, I just realized that by copying dictionary's implementation, I cut out the call site to this. I'll take another look and either add back the call site or delete the resource.

Reply:
BTW, we do try to protect against corruption from OOM ("// Assign member variables after both arrays are allocated to guard against corruption from OOM if second fails."). Which is unusual, but I could imagine the value -- you put a lot of things in the dictionary, maybe it's important to you 🙂 -- but we don't protect against corruption from trying to go past the highest prime number. We should probably do neither or both.

Reply:
Not suggesting this isa concern for today.

Reply:
> I expect that if you try to add to many items to a dictionary it likely simply corrupts itself.

It won't. There is another check inside HashHelpers.GetPrime that prevents the corruption. Here is the exception that you will get if you try adding items to Dictionary until it hits the limit:
```
Unhandled exception. System.ArgumentException: Hashtable's capacity overflowed and went negative. Check load factor, capacity and the current size of the table.
   at System.Collections.HashHelpers.GetPrime(Int32 min)
   at System.Collections.HashHelpers.ExpandPrime(Int32 oldSize)
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
```
It seems to work as expected except a minor nit that the exception message talks about Hashtable and load factor (that does not exist for Dictionary).

Reply:
Yeah, I checked that as well and deleted the unused resource last night, though I didn't notice the inconsistent message; I can update that resource as well.